Migrate to Qt5

liberostelios opened the pull request:
This is a migration of Moka to Qt5. It is tested and working almost perfectly, except with an issue against HiDPI screens where picking darts does not work as intended (probably related to the way a mouse click is translated to a ray?).

(The request also contains two commits for some fixes for which there was another PR submitted before.)

liberostelios committed:
Fix building issue on OSX

liberostelios committed:
Temporarily fix an issue with GLWindow in OSX

Fixed by disable the drawing of selection, as the QPainter was
conflicting with the GLWindow.

liberostelios committed:
Migrate to Qt5 (without build)

This is not building due to a linking issue, but the code must be
fine.

liberostelios committed:
Complete migration to Qt5

liberostelios committed:
Fix an issue related to Qt5

gdamiand commented:
It seems this PR includes the previous one. Is it true ? If yes I can remove the first PR.

gdamiand commented on src/mokaQtIhm/windows/gl-window.qt.cc:
```diff
@@ -127,7 +127,7 @@ void GLWindow::closeEvent(QCloseEvent * e)
 void GLWindow::paintGL()
 {
-  QPainter p(this);
+//  QPainter p(this);
```
Instead of commenting all these code, it is better to protect it by #ifndef directives, so that by default this code is here, and we can remove it by defining an option in CMake.
About naming, I suggest DISABLE_SELECTION_BOX (but feel free to suggest any better name if you have some idea).

liberostelios commented:
Indeed, I submitted two different PRs in case you wanted to only proceed with the fixes for now and let the Qt5 merge for later on.

liberostelios committed:
Add a CMake option to disable the selection box

liberostelios committed:
Merge branch 'master' into qt5

liberostelios commented:
I implemented the CMake option for disabling the selection box, which is indeed much cleaner solution.

gdamiand committed:
Update src/mokaQtIhm/CMakeLists.txt to use last cmake policy.

gdamiand commented:
There are some missing shortcuts; plus some warning when you run moka about some problem with some QObject::connect.

These problems must be solved before to merge this pr.

gdamiand commented on src/mokaQtIhm/object-options/options-couleurs.qt.cc:
```diff
@@ -105,9 +106,9 @@ OptionCouleurs :: OptionCouleurs(Window * parent)
-   Raccourci -> connectItem(2 , this , SLOT(close())) ;
+//   Q3Accel * Raccourci = new Q3Accel(this) ;
+//   Raccourci -> insertItem(QKeySequence(Qt::Key_Space) , 2) ;
+//   Raccourci -> connectItem(2 , this , SLOT(close())) ;
```
This shortcut must be re-added.

gdamiand commented on src/mokaQtIhm/object-options/options-frame.qt.cc:
```diff
@@ -63,49 +63,49 @@ OptionsFrame :: OptionsFrame ( Window * parent )
-  Raccourci -> connectItem ( 14 , this , SLOT (callbackTournerButton() ))  ;
+//  Raccourci -> insertItem ( QKeySequence ( "Alt+F11" ) , 14 ) ;
+//  Raccourci -> connectItem ( 14 , this , SLOT (callbackTournerButton() ))  ;
   
```
All these shortcuts must be re-added.

liberostelios replied:
I think some of those shortcuts (those regarding the tabs of the options window) are not really useful, since the main-window shortcuts are overriding them and provide the same functionality anyway. But I guess it doesn't harm to keep them.

gdamiand commented on src/mokaQtIhm/object-options/options-volumic-homology.qt.cc:
```diff
@@ -48,9 +48,9 @@ OptionsVolumicHomology :: OptionsVolumicHomology(Window * parent) :
+//   Q3Accel * Raccourci = new Q3Accel(this) ;
+//   Raccourci -> insertItem(QKeySequence(Qt :: Key_Space) , 9) ;
+//   Raccourci -> connectItem(9 , this , SLOT(close()))  ;
 
```
This shortcut must be re-added.

liberostelios committed:
Fix issues with warning regarding shortcuts

liberostelios committed:
Migrate all previous shortcuts from Q3Accel to QShortcuts

liberostelios committed:
Re-enable all shortcuts of the main window

liberostelios committed:
Refactoring of leftovers from the Q3Accel references

liberostelios commented:
I believe with those commits, all shortcuts are working again. I also fixed the issues with the connect calls of some shortcuts.

gdamiand committed:
Bugfix in move selection operations.

gdamiand commented:
Thanks @liberostelios.
Here some (last?) bugs:
* space does not close the creation mode (while it should be)
* same for translation/rotation/scale mode (space should close the operation)
* multi-view is broken (shortcut 4)

Expect these last problems, I tested everything and it seems ok.
(by the way I also corrected two bugs, and push in your branch).

liberostelios committed:
Fix issue with ASSIMP include dir variable

liberostelios committed:
Merge branch 'qt5' of github.com:liberostelios/moka into qt5

liberostelios committed:
Fix an issue with space-related shortcuts

The hideAllWindow shortcut on the main Window would consume the keySpace
shortcut, breaking any other allocation of the shortcut to actions.
Temporarily, it's been disabled in order to allow shortcuts on other operations
(creation, rotation etc.) to work with the space key.

liberostelios commented:
There seems to be a problem with Qt5 which breaks the current keyboard shortcuts combination.

Apparently, Space cannot be both the shortcut for "callbackHideAllWindow()" on the main Window and each individual dialog. I suppose Qt5 consumes any keyboard event only once and runs it for the first shortcut definition it finds. The only solution is to assign different shortcuts for those behaviour in order to avoid the conflict. For now, I have disabled the "Hide All Windows" shortcut, so Space now indeed closes the creation mode and other operations.

Still investigating the multi-view issue, which is also very strange. And, now, with Qt5.11 I have an issue where the GUI is completely broken on Retina/HiDPI screens.

liberostelios committed:
Enforce c++11 standard in cmake
